### ISSUE TITLE
Completion is really slow in  ".cpp" file while it's normal in ".c" file.

### ISSUE THREAD

**Issue by moyotar** [company-mode/company-mode#592]
> This is my config.

```
(global-company-mode 1)
(setq company-idle-delay 0.05)
(setq company-minimum-prefix-length 2)
;; set default `company-backends'
(setq company-backends
      '((company-files          ; files & directory
         company-keywords       ; keywords
         company-capf
         company-yasnippet
         )
        (company-abbrev company-dabbrev)
        ))
(dolist (hook '(c-mode-hook
				c++-mode-hook
				))
  (add-hook hook
			(lambda ()
			  (add-to-list (make-local-variable 'company-backends)
						   '(company-clang))
			  )
			)
  )
```

> Problem:

1. eg: After I typed `pr`, It wasted several seconds before showing candidates like `printf`. 
2. Ignoring the responsiveness, sometimes It can show candidates and sometimes can't.

**Comment by dgutov** [comment]
There is not much we can do here, sorry: `company-clang` uses the `clang` program for completion, and it's slow in nontrivial C++ projects (C++ is generally hard to parse, the standard library is larger, etc).

Please try one of the alternative backends mentioned here: https://github.com/company-mode/company-mode/wiki/Third-Party-Packages#cc

**Comment by moyotar** [comment]
@dgutov 
In `.cpp` fileI found a frightening truth that after typing `pr`,  the `*clang-output*` buffer output 3M message before showing candidates. I think that is why it needs a few seconds. Do others using `company-clang`  in `c++` have  to accept it’s speed to completion? Or they have some config to speed it up?

**Comment by moyotar** [comment]
It's in a really small project which only contains several sourse files.

**Comment by dgutov** [comment]
> In .cpp file, I found a frightening truth that after typing pr, the *clang-output* buffer output 3M message before showing candidates. I think that is why it needs a few seconds.

Have you tried calling `clang` with the same arguments from the command line? Is it much faster that way?

**Comment by moyotar** [comment]
How to use clang ni command line to do the same thing? What arguments?

**Comment by moyotar** [comment]
Does `company-clang-arguments` help something? Its value is nil.

**Comment by dgutov** [comment]
If you have a `*clang-error*` buffer, the command is at the top. It looks something like this:

```
/usr/bin/clang -fsyntax-only -Xclang -code-completion-macros -x c++ -Xclang -code-completion-at=-:43:3 -
```

The dashes designate the fact that the file contents are sent via stdin. You can also pass the file name to it.
